### PR TITLE
feat: audio block ux review

### PR DESCRIPTION
**Pull request by ragi96**
Fixes of the audio block UX Review [Task](https://app.clickup.com/t/ndbnjd)

Some tasks within the task are related to attachments this are handled in another PR #506

### CODE REVIEW

File: packages/audio-block/src/AudioBlock.tsx [hunk 1]
```diff
@@ -103,6 +103,9 @@ export const AudioBlock = ({ appBridge }: BlockProps) => {
     };
 
     const updateAudioAsset = async (audio: Asset) => {
+        if (title === undefined) {
```
Comment (celine-s) [inline packages/audio-block/src/AudioBlock.tsx:114]:
```suggestion
        if (!title) {
```

File: packages/audio-block/src/AudioBlock.tsx [hunk 2]
```diff
@@ -103,6 +103,9 @@ export const AudioBlock = ({ appBridge }: BlockProps) => {
...
+            saveTitle(`[{"type":"heading3","children":[{"text":"${audio.title}"}]}]`);
```
[the remark this answers is not in the document]
Comment (ragi96) [reply packages/audio-block/src/AudioBlock.tsx:107]: afaik there is atm no option for that, if there is one i am happy to remove that ugly string

File: packages/audio-block/src/AudioBlock.tsx [hunk 3]
```diff
@@ -186,32 +190,38 @@ export const AudioBlock = ({ appBridge }: BlockProps) => {
...
+                    {description !== DEFAULT_CONTENT_DESCRIPTION || isEditing ? (
+                        <div data-test-id="audio-block-description">
+                            <RichTextEditor
+                                designTokens={designTokens}
+                                border={false}
+                                position={Position.FLOATING}
+                                onBlur={saveDescription}
+                                placeholder={isEditing ? 'Add a description here' : undefined}
+                                readonly={!isEditing}
+                                value={description ?? DEFAULT_CONTENT_DESCRIPTION}
```
Comment (celine-s) [inline packages/audio-block/src/AudioBlock.tsx]: the type on line 44 is the wrong one. It is just `p` not `paragraph` => `const DEFAULT_CONTENT_DESCRIPTION = '[{"type":"p","children":[{"text":""}]}]';`

File: packages/audio-block/src/AudioBlock.tsx [hunk 4]
```diff
@@ -186,32 +190,38 @@ export const AudioBlock = ({ appBridge }: BlockProps) => {
...
-                            designTokens={designTokens}
-                            border={false}
-                            position={Position.FLOATING}
-                            onBlur={saveDescription}
-                            placeholder={isEditing ? 'Add a description here' : undefined}
-                            readonly={!isEditing}
-                            value={description ?? DEFAULT_CONTENT_DESCRIPTION}
-                        />
-                    </div>
+                    {title !== DEFAULT_CONTENT_TITLE || isEditing ? (
```
Comment (celine-s) [inline packages/audio-block/src/AudioBlock.tsx:201]: this conditions could be simplified, when using `serializeRawToHtml`. See e.g. in the Image Block: Image Caption

File: packages/audio-block/src/AudioBlock.tsx [hunk 5]
```diff
@@ -186,32 +190,38 @@ export const AudioBlock = ({ appBridge }: BlockProps) => {
...
+
+                    {description !== DEFAULT_CONTENT_DESCRIPTION || isEditing ? (
+                        <div data-test-id="audio-block-description">
+                            <RichTextEditor
+                                designTokens={designTokens}
+                                border={false}
+                                position={Position.FLOATING}
+                                onBlur={saveDescription}
+                                placeholder={isEditing ? 'Add a description here' : undefined}
+                                readonly={!isEditing}
```
[the remark this answers is not in the document]
Comment (ragi96) [reply packages/audio-block/src/AudioBlock.tsx]: Changed the handling for the rte's 👍

File: packages/audio-block/src/AudioBlock.tsx [hunk 6]
```diff
@@ -186,32 +190,38 @@ export const AudioBlock = ({ appBridge }: BlockProps) => {
...
+                                value={description ?? DEFAULT_CONTENT_DESCRIPTION}
+                            />
+                        </div>
+                    ) : null}
                 </div>
                 {audio ? (
                     <div className="tw-flex tw-gap-2">
-                        <DownloadButton onDownload={() => downloadAsset(audio)} />
+                        {downloadable ? <DownloadButton onDownload={() => downloadAsset(audio)} /> : null}
```
[the remark this answers is not in the document]
Comment (ragi96) [reply packages/audio-block/src/AudioBlock.tsx]: adjusted

File: packages/audio-block/src/AudioBlock.tsx [hunk 7]
```diff
@@ -186,32 +190,38 @@ export const AudioBlock = ({ appBridge }: BlockProps) => {
...
+                    ) : null}
+
+                    {description !== DEFAULT_CONTENT_DESCRIPTION || isEditing ? (
+                        <div data-test-id="audio-block-description">
+                            <RichTextEditor
+                                designTokens={designTokens}
+                                border={false}
+                                position={Position.FLOATING}
+                                onBlur={saveDescription}
+                                placeholder={isEditing ? 'Add a description here' : undefined}
```
[the remark this answers is not in the document]
Comment (ragi96) [reply packages/audio-block/src/AudioBlock.tsx]: adjusted

File: packages/audio-block/src/AudioBlock.tsx [hunk 8]
```diff
@@ -186,32 +190,38 @@ export const AudioBlock = ({ appBridge }: BlockProps) => {
...
-                            value={description ?? DEFAULT_CONTENT_DESCRIPTION}
-                        />
-                    </div>
+                    {title !== DEFAULT_CONTENT_TITLE || isEditing ? (
+                        <div data-test-id="audio-block-title">
+                            <RichTextEditor
+                                designTokens={designTokens}
+                                border={false}
+                                onBlur={saveTitle}
+                                placeholder={isEditing ? 'Asset name' : undefined}
```
[the remark this answers is not in the document]
Comment (ragi96) [reply packages/audio-block/src/AudioBlock.tsx]: adjusted

File: packages/audio-block/src/AudioBlock.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (celine-s) [reply packages/audio-block/src/AudioBlock.tsx:115]: I thought I had the same issue on the Image Block and has a fix. But it looks a bit broken atm

**Review by celine-s — COMMENTED** [review]
just some small stuff & the RTE part. Rest looks really good. Love the block - nice work!!

File: packages/audio-block/src/AudioBlock.tsx [hunk 9]
```diff
@@ -186,32 +198,58 @@ export const AudioBlock = ({ appBridge }: BlockProps) => {
...
+                                    plugins={customTitlePlugins}
+                                    updateValueOnChange
+                                />
+                            )}
+                        </div>
+                    ) : null}
+
+                    {description !== DEFAULT_CONTENT_DESCRIPTION || isEditing ? (
+                        <div>
+                            {!isEditing ? (
```
Comment (celine-s) [inline packages/audio-block/src/AudioBlock.tsx:228]: do you need both conditions & the `hasRichTextValue(description)`?

File: packages/audio-block/src/AudioBlock.tsx (hunk 1 above)
Comment (ragi96) [reply packages/audio-block/src/AudioBlock.tsx]: adjusted

File: packages/audio-block/src/AudioBlock.tsx (hunk 3 above)
Comment (ragi96) [reply packages/audio-block/src/AudioBlock.tsx]: oops, where did i get that from, adjusted

File: packages/audio-block/src/AudioBlock.tsx (hunk 4 above)
Comment (ragi96) [reply packages/audio-block/src/AudioBlock.tsx]: cleaned the rte should be better now

File: packages/audio-block/src/AudioBlock.tsx (hunk 9 above)
Comment (ragi96) [reply packages/audio-block/src/AudioBlock.tsx]: no for sure not, removed the bad ones

File: packages/audio-block/src/AudioBlock.tsx [hunk 10]
```diff
@@ -186,32 +198,46 @@ export const AudioBlock = ({ appBridge }: BlockProps) => {
...
-                            readonly={!isEditing}
+                            placeholder="Asset name"
                             value={title ?? DEFAULT_CONTENT_TITLE}
                             plugins={customTitlePlugins}
+                            updateValueOnChange
                         />
-                    </div>
-                    <div data-test-id="audio-block-description">
+                    )}
+                    {!isEditing ? (
```
Comment (celine-s) [inline packages/audio-block/src/AudioBlock.tsx:218]: would do it the other way around: `{isEditing ? <RichTextEditor/> : html}`. Anyway it looks now much clearer!

**Review by celine-s — APPROVED** [review]

File: packages/audio-block/src/AudioBlock.tsx (hunk 9 above)
Comment (celine-s) [reply packages/audio-block/src/AudioBlock.tsx]: Looks good like this, also in the UI.

File: packages/audio-block/src/AudioBlock.tsx (hunk 10 above)
Comment (ragi96) [reply packages/audio-block/src/AudioBlock.tsx]: 💯 adjusted

File: packages/audio-block/src/AudioBlock.tsx [hunk 11]
```diff
@@ -186,32 +198,46 @@ export const AudioBlock = ({ appBridge }: BlockProps) => {
...
+                            {hasRichTextValue(description) && (
+                                <div
+                                    data-test-id="block-description-html"
+                                    dangerouslySetInnerHTML={{ __html: htmlDescription }}
+                                />
+                            )}
+                        </>
+                    )}
                 </div>
                 {audio ? (
```
Comment (celine-s) [inline packages/audio-block/src/AudioBlock.tsx:238]: audio &&

Comment (ragi96) [reply packages/audio-block/src/AudioBlock.tsx]: adjusted 👍

File: packages/audio-block/src/AudioBlock.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ragi96) [reply packages/audio-block/src/AudioBlock.tsx:115]: Probably we should open a new task to improve that handling.

[the remark this answers is not in the document]
Comment (triggertoo) [reply packages/audio-block/src/AudioBlock.tsx:115]: sounds good! could you create on in clickup?

[the remark this answers is not in the document]
Comment (ragi96) [reply packages/audio-block/src/AudioBlock.tsx:115]: [Created](https://app.clickup.com/t/8677nh1xj)

**Review by triggertoo — APPROVED** [review]
looks good 👍

**Pull request merged**